astropy/units/structured.py

```diff
@@ -487,6 +487,12 @@ def __ne__(self, other):
 
         return self.values() != other.values()
 
+    def __reduce_ex__(self, protocol):
```

Comment:
I think a simpler solution is just to pass `__new__` something that will actually give the desired uninitialized `StructuredUnit`. The way you have it, the code goes through all the trouble of initializing the unit from a string, and then override the result will the old state.

I check, and the following replacement also passes all the test:
```
    def __getnewargs__(self):
        return (), None
```

Actually, reading the description of those methods at https://docs.python.org/3/library/pickle.html, it is mentioned that those are used by `copy.copy` as well - and indeed that does not work without `__getnewargs__`...  I'll make a quick commit to fix that too...
Basically this does `StructuredUnit(units=())`, which returns an empty item.

I considered just making that the default in `__new__` but then one gets a unit that cannot round-trip through `u.Unit(str(u.StructuredUnit(()))`, so seems less than ideal.

Reply:
Looks like a more elegant solution! And it's nice to pick up ``copy`` support. Thanks!